cardano-api/src/Cardano/Api.hs

```diff
@@ -95,6 +101,37 @@ shelleyGenSigningKey :: IO SigningKey
 shelleyGenSigningKey =
     SigningKeyShelley . Shelley.SKey <$> runSecureRandom genKeyDSIGN
 
+-- | Register a shelley staking key.
+shelleyRegisterStakingKey
+  :: (SigningKey, VerificationKey)
+  -> Certificate
+shelleyRegisterStakingKey keyPair = do
+  let cred = mkShelleyCredential keyPair
+  ShelleyDelegationCertificate $ Shelley.DCertDeleg $ Shelley.RegKey cred
+
+-- | Deregister a shelley staking key.
+shelleyDeregisterStakingKey
+  :: (SigningKey, VerificationKey)
+  -> Certificate
+shelleyDeregisterStakingKey keyPair = do
+  let cred = mkShelleyCredential keyPair
+  ShelleyDelegationCertificate $ Shelley.DCertDeleg $ Shelley.DeRegKey cred
+
+-- | Delegate your stake (as the delegator) to a specified delegatee.
+shelleyDelegateStake
+  :: (SigningKey, VerificationKey)
+  -> VerificationKey
+  -> Certificate
+shelleyDelegateStake delegatorKeyPair (VerificationKeyShelley delegateeVKey) = do
+  let cred = mkShelleyCredential delegatorKeyPair
+  ShelleyDelegationCertificate $ Shelley.DCertDeleg $ Shelley.Delegate $ Shelley.Delegation cred (Shelley.hashKey delegateeVKey)
+shelleyDelegateStake _ _ = panic "Cardano.Api.shelleyDelegateStake Please use a shelley key pair."
+
+mkShelleyCredential :: (SigningKey, VerificationKey) -> ShelleyCredential
+mkShelleyCredential (SigningKeyShelley sKey, VerificationKeyShelley vKey) =
+  Shelley.toCred $ Shelley.KeyPair vKey sKey
```

Comment:
This should bypass `toCred` and do it directly, without needing the signing key. The signing key is not used by `toCred` and we do not want to have to pass it. We only need the verification key hash.

So instead of ` (SigningKey, VerificationKey)` it should be `VerificationKeyHash`, and same for the the stake pool arg in `shelleyDelegateStake`. For stake pools especially we only get the key hash, not the key. The key hash is used as the identifier.